src/Abp.Zero/Authorization/Users/AbpUserStore.cs

```diff
@@ -278,18 +284,23 @@ public virtual async Task RemoveFromRoleAsync(TUser user, string roleName)
         [UnitOfWork]
         public virtual async Task<IList<string>> GetRolesAsync(TUser user)
         {
-            var query = from userRole in _userRoleRepository.GetAll()
-                        join role in _roleRepository.GetAll() on userRole.RoleId equals role.Id
-                        where userRole.UserId == user.Id
-                        select role.Name;
+            var userRoles = from userRole in _userRoleRepository.GetAll()
+                            join role in _roleRepository.GetAll() on userRole.RoleId equals role.Id
+                            where userRole.UserId == user.Id
+                            select role.Name;
 
-            return await AsyncQueryableExecuter.ToListAsync(query);
+            var userOrganizationUnitRoles = from userOu in _userOrganizationUnitRepository.GetAll()
+                                            join roleOu in _roleOrganizationUnitRepository.GetAll() on userOu.OrganizationUnitId equals roleOu.OrganizationUnitId
+                                            join userOuRoles in _roleRepository.GetAll() on roleOu.RoleId equals userOuRoles.Id
+                                            where userOu.UserId == user.Id
+                                            select userOuRoles.Name;
+
+            return await AsyncQueryableExecuter.ToListAsync(userRoles.Concat(userOrganizationUnitRoles));
```

Comment:
Do we need to use `Distinct()` on the resulting role names?

Reply:
Right